packages/ingest/queue.js

```diff
@@ -1,111 +1,116 @@
+  const message = buildQueueMessageFromTemplate({
+    collection,
+    messageTemplate,
+    parentExecutionArn,
+    payload,
+    provider,
+    queueName
+  });
 
-  message.cumulus_meta.execution_name = uuidv4();
   const arn = getExecutionArn(
     message.cumulus_meta.state_machine,
     message.cumulus_meta.execution_name
   );
+
   await sendSQSMessage(queueUrl, message);
+
   return arn;
 }
 module.exports.enqueueParsePdrMessage = enqueueParsePdrMessage;
 
 /**
  * Enqueue a granule to be ingested
  *
- * @param {Object} granule - the granule to be enqueued for ingest
- * @param {string} queueUrl - the SQS queue to add the message to
- * @param {string} granuleIngestMessageTemplateUri - the S3 URI of template for
+ * @param {Object} params
+ * @param {Object} params.granule - the granule to be enqueued for ingest
+ * @param {string} params.queueUrl - the SQS queue to add the message to
+ * @param {string} params.granuleIngestMessageTemplateUri - the S3 URI of template for
  * a granule ingest message
- * @param {Object} provider - the provider config to be attached to the message
- * @param {Object} collection - the collection config to be attached to the
+ * @param {Object} params.provider - the provider config to be attached to the message
+ * @param {Object} params.collection - the collection config to be attached to the
  *   message
- * @param {Object} pdr - an optional PDR to be configured in the message payload
- * @param {string} parentExecutionArn - parent workflow execution arn to add to the message
+ * @param {Object} params.pdr - an optional PDR to be configured in the message payload
+ * @param {string} params.parentExecutionArn - parent workflow execution arn to add to the message
  * @returns {Promise} - resolves when the message has been enqueued
  */
-async function enqueueGranuleIngestMessage(
+async function enqueueGranuleIngestMessage({
   granule,
   queueUrl,
   granuleIngestMessageTemplateUri,
   provider,
   collection,
   pdr,
   parentExecutionArn
-) {
-  // Build the message from a template
-  const message = await getMessageFromTemplate(granuleIngestMessageTemplateUri);
+}) {
+  const messageTemplate = await getMessageFromTemplate(granuleIngestMessageTemplateUri);
 
-  message.payload = {
+  // Should this throw? Probably not because it would prevent all the other messages
+  // from being queued, since this function is invoked via Promise.all in queue-granules
+  const queueName = getQueueNameByUrl(messageTemplate, queueUrl);
+
+  const payload = {
     granules: [
       granule
     ]
   };
-  if (pdr) message.meta.pdr = pdr;
 
-  message.meta.provider = provider;
-  message.meta.collection = collection;
-  if (parentExecutionArn) message.cumulus_meta.parentExecutionArn = parentExecutionArn;
+  const message = buildQueueMessageFromTemplate({
+    collection,
+    messageTemplate,
+    parentExecutionArn,
+    payload,
+    provider,
+    queueName
+  });
+
+  if (pdr) message.meta.pdr = pdr;
```

Comment:
The fact that `pdr` is a one-off that isn't taken care of by previous logic is somewhat alarming to me - only because I'm worried that there are others. Are there any other values that you can think of that we'd need to add ad-hoc logic for?

Integration tests probably would catch any missing situations...

Reply:
nvm this question doesn't make sense.